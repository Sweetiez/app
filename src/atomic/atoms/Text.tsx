import * as React from 'react';
import styled from 'styled-components';
import colors from '../../assets/colors';

interface Props {
  content: string;
  numberOfLines?: number;
  size?: number;
  color?: string;
  textDecoration?: string;
  textTransform?: string;
  textAlign?: string;
}

const StyledText = styled.Text<{
  size;
  color;
  textTransform;
  textDecoration;
  textAlign;
}>`
  font-family: Pompiere-Regular;
  color: ${({color}) => color};
  font-size: ${({size}) => size}px;
  text-decoration: ${({textDecoration}) => textDecoration};
  text-transform: ${({textTransform}) => textTransform};
  text-align: ${({textAlign}) => textAlign};
`;

const Text: React.FC<Props> = ({
  content,
  numberOfLines = undefined,
  size = 12,
  color = colors.black,
  textDecoration = 'none',
  textTransform = 'none',
  textAlign = 'left',
}) => (
  <StyledText
    numberOfLines={numberOfLines}
    size={size}
    color={color}
    textDecoration={textDecoration}
    textAlign={textAlign}
    textTransform={textTransform}>
    {content}
  </StyledText>
);

export default Text;
