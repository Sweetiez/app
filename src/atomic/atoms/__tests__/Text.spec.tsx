import * as React from 'react';
import renderer from 'react-test-renderer';
import Text from '../Text';

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Text content={'test'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with smaller size', () => {
    const tree = renderer.create(<Text content={'test'} size={10} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with specified number of lines', () => {
    const tree = renderer
      .create(
        <Text
          content={
            'test renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines test'
          }
          numberOfLines={3}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
