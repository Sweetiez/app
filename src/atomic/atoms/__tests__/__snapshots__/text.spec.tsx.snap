// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  color="#000000"
  size={12}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Pompiere-Regular",
        "fontSize": 12,
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
        "textTransform": "none",
      },
    ]
  }
  textDecoration="none"
  textTransform="none"
>
  test
</Text>
`;

exports[`Text renders with smaller size 1`] = `
<Text
  color="#000000"
  size={10}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Pompiere-Regular",
        "fontSize": 10,
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
        "textTransform": "none",
      },
    ]
  }
  textDecoration="none"
  textTransform="none"
>
  test
</Text>
`;

exports[`Text renders with specified number of lines 1`] = `
<Text
  color="#000000"
  numberOfLines={3}
  size={12}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Pompiere-Regular",
        "fontSize": 12,
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
        "textTransform": "none",
      },
    ]
  }
  textDecoration="none"
  textTransform="none"
>
  test renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines renders with specified number of lines test
</Text>
`;
